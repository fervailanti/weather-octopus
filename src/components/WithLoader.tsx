import React, { FC, ReactElement } from 'react'
import Loader from './Loader'

interface WithLoaderProps {
  isLoading?: boolean
  className?: string
}

const WithLoader: FC<WithLoaderProps> = ({
  isLoading = false,
  className = '',
  children
}): ReactElement => {
  const visibility = {
    content: isLoading ? 'invisible' : 'visible',
    loader: isLoading ? 'visible' : 'invisible'
  }
  return (
    <>
      <div className={`${visibility.content} ${className}`}>{children}</div>
      <div className={`${visibility.loader} absolute right-1/2 top-1/2 -mr-10 -mt-10`}>
        <Loader />
      </div>
    </>
  )
}

export default WithLoader
