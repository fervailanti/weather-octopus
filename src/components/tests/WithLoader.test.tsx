import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import WithLoader from '../WithLoader'

const mockContent = <p>Hi! I'm a content</p>

describe('WithLoader High Order Component', () => {
  it.each([
    [
      "hide the content and show the spinner if it's loading",
      <WithLoader isLoading={true}>{mockContent}</WithLoader>,
      'invisible'
    ],
    [
      "hide the spinner and show the content if it's not loading",
      <WithLoader isLoading={false}>{mockContent}</WithLoader>,
      'visible'
    ]
  ])('Should %s', (_, component, expected) => {
    const withLoader = render(component)
    expect(withLoader.container).toMatchSnapshot()
    expect(withLoader.container.firstChild).toHaveClass(expected)
  })
})
