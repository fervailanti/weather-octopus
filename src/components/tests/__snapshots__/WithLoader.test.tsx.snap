// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithLoader High Order Component Should hide the content and show the spinner if it's loading 1`] = `
<div>
  <div
    class="invisible "
  >
    <p>
      Hi! I'm a content
    </p>
  </div>
  <div
    class="visible absolute right-1/2 top-1/2 -mr-10 -mt-10"
  >
    <div
      class="loader"
    >
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`WithLoader High Order Component Should hide the spinner and show the content if it's not loading 1`] = `
<div>
  <div
    class="visible "
  >
    <p>
      Hi! I'm a content
    </p>
  </div>
  <div
    class="invisible absolute right-1/2 top-1/2 -mr-10 -mt-10"
  >
    <div
      class="loader"
    >
      <div />
      <div />
    </div>
  </div>
</div>
`;
